modules/gojira-core/src/test/java/com/flipkart/gojira/core/AspectjMethodInterceptionTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ *
+ * Copyright 2020 Flipkart Internet, pvt ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http:www.apache.orglicensesLICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.flipkart.gojira.core;
+
+import com.flipkart.gojira.core.annotations.ProfileOrTest;
+import com.flipkart.gojira.core.aspect.test.AspectJConfig;
+import com.flipkart.gojira.core.aspect.test.AspectJUnit4Runner;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertEquals;
+
+@AspectJConfig(classpathAdditions = "src/test/java/com/flipkart/gojira/core/aspect/test")
+@RunWith(AspectJUnit4Runner.class)
+public class AspectjMethodInterceptionTest {
+
+    @ProfileOrTest
+    private int testPrivateMethod (int i) {
+        return i;
+    }
+
+    private int testNonAnnotated (int i) {
+        return i;
+    }
+
+    @Test
+    public void testMethodInterception () throws Throwable {
+        assertEquals(11, testPrivateMethod(10));
+    }
+
+    @Test
+    public void testMethodInterception_nonAnnotated () throws Throwable {
+        assertEquals(11, testNonAnnotated(10));
```

Comment:
Wondering how is aspect/method interception working on a non annotated method! expected result should be 10 only right?

Reply:
In aop.xml we have added a concrete-aspect  and defined the pointcut to be within this test class. So any method execution in that (package or class) will be intercepted

Reply:
We totally have two aspects in the test. One which is annotation based. And one which specifies the location of the classes. All method executions in that location will be intercepted irrespective of annotation.